bitcoin: Update to 0.19.1

fanquake opened the pull request:
## Description
Update Bitcoin Core to 0.19.1
Enable hardening.
Protobuf is no longer required.
Use Python 3.8.
`--disable-experimental-asm` was renamed `--enable-asm` and is used by default.

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on
macOS 10.14.6 18G2022
Xcode 11.3.1 11C504

###### Verification <!-- (delete not applicable items) -->
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?
<!-- Please don't open a new Trac ticket if you are submitting a pull request. -->
- [ ] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

reneeotten commented on finance/bitcoin/Portfile:
```diff
@@ -1,11 +1,10 @@
 categories              finance crypto
-version                 0.17.1
+version                 0.19.0.1
 revision                3
```
the ```revision``` needs to be reset to ```0``` when updating the version

fanquake replied:
Thanks, have fixed that up.

tobypeterson commented:
Consistent build failures...

fanquake commented:
> Consistent build failures...

As per the checkboxes in the op, I haven't been testing building or installing; I'm not a macports user. 

If anything, this Pull Request is to highlight the fact that this port is very outdated, and being built with nonsensical options, such as `--disable-hardening`. I'd be just as happy to see it removed given it's current state.

tobypeterson commented:
Ah, that makes more sense. Perhaps this PR will motivate someone to fix up the port; the error just seems like a missing dependency, may not be that hard.

pmetzger commented:
@easye Are you still maintaining this port?

fanquake commented:
There's been another release in the mean time (0.19.1). So I've rebased this to fix the merge conflicts, and updated for that. Also added a change that I think will solve the bdb related build issues.

easye commented:
Sorry, but my attention has been spread around quite a bit.  

This  port is openmaintainer, so anyone with commit rights can push the results so far.  

Before I commit things, I usually try to build it at least once.  I'll try to get some cycles over the next day to build this for ya…

easye commented:
@fanquake Thanks for the solid work!

N.b. unaudited other than builds under macOS 10.15.x Catalina by @easye
